gossipd/routing.c

```diff
@@ -994,3 +1001,86 @@ struct route_hop *get_route(tal_t *ctx, struct routing_state *rstate,
 	/* FIXME: Shadow route! */
 	return hops;
 }
+
+/* Get a singleton array of struct node_connection containing
+ * the channel matching the short_channel_id, which must be
+ * an out connection of the given node. */
+static struct node_connection **
+get_out_node_connection_of(const tal_t *ctx,
+			   struct routing_state *rstate,
+			   const struct node *node,
+			   const struct short_channel_id *short_channel_id)
+{
+	struct node_connection **ncs;
+	int i;
+
+	for (i = 0; i < tal_count(node->out); ++i) {
+		if (structeq(&node->out[i]->short_channel_id, short_channel_id)) {
+			ncs = tal_arr(ctx, struct node_connection*, 1);
+			ncs[0] = node->out[i];
+			return ncs;
+		}
+	}
+
+	return NULL;
+}
+/* Get an array of all channels connected to this node. */
+static struct node_connection **
+get_all_node_connections_of(const tal_t *ctx,
+			    struct routing_state *rstate,
+			    const struct node *node)
+{
+	struct node_connection **ncs;
+
+	u64 num_ins = tal_count(node->in);
+	u64 num_outs = tal_count(node->out);
+	u64 num = num_ins + num_outs;
```

Comment:
tal_count returns size_t, which is pretty natural for sizes; u64 looks a bit weird here.